enhancements/network/allow-from-router-networkpolicy.md

```diff
@@ -0,0 +1,253 @@
+---
+title: "Allow-From-Router" NetworkPolicy
+authors:
+  - "@danwinship"
+reviewers:
+  - "@squeed"
+  - "@trozet"
+  - "@Miciah"
+approvers:
+  - "@knobunc"
+  - "@squeed"
+creation-date: 2020-12-10
+last-updated: 2020-12-10
+status: implementable
+---
+
+# "Allow-From-Router" NetworkPolicy
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [X] Operational readiness criteria is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Many users want to create NetworkPolicies that say, effectively,
+"allow traffic from routers". However, the router can run with one of
+several different "endpoint publishing strategies", which use the
+cluster network differently, and when using the `HostNetwork`
+strategy, the NetworkPolicy-relevant behavior also varies depending on
+the network plugin in use. This makes it impossible to create an
+"allow traffic from routers" policy that will work in any cluster.
+Indeed, in some cases, it is difficult to create such a policy even
+just for a single cluster, even when you know exactly how the cluster
+is configured.
+
+## Motivation
+
+### Goals
+
+- Allow a user to create a policy which will allow traffic to their
+  pods from routers (and not from arbitrary pod-network pods), which
+  will work regardless of network plugin or router endpoint publishing
+  strategy.
+
+### Non-Goals
+
+- Allowing a user to create a policy which is 100% guaranteed to allow
+  traffic _only_ from routers (and, eg, not from other host-network
+  pods).
+
```

Comment:
Users have also expressed interest in allowing host-network pods with NetworkPolicy. Would exposing this without reference to the router be a goal or non-goal of this Enhancement?

Reply:
(It seems like this is mostly a question of "are we publishing this as a documented API commitment?")

Reply:
That's neither a goal nor a non-goal, though it's basically what I propose. (Assuming you mean "host-network in general", not "I want to allow from host-network pod X specifically, but not from all nodes", which is definitely a non-goal.)

Reply:
Sorry, replied in a "stale" window and didn't see your second comment. Yes, I think if we actually have a visible `policy-group: host-network` then that would have to be a documented feature on its own.